SetupDataPkg/Tools/GenCfgData.py

```diff
@@ -1560,15 +1560,93 @@ def generate_delta_svd_from_bin(self, old_data, new_data):
             offset += int(exec['CfgHeader']['length'], 0)
             offset += int(exec['CondValue']['length'], 0)
-            bytes_array.append(bytes[offset:])
+            cfg_hdr = self.get_item_by_index(exec["CfgHeader"]["indx"])
+            tag_val = array_str_to_value(cfg_hdr["value"]) >> 20
+            name = "Device.ConfigData.TagID_%08x" % tag_val
+            var = UEFIVariable(name, uuid.UUID(SETUP_CONFIG_POLICY_VAR_GUID), bytes[offset:], attributes=3)
+            buf = create_vlist_buffer(var)
+            bytes_array.append(buf)
+            name_array.append(name)
+
+        return (name_array, bytes_array)
+
+    def get_var_by_index(self, index):
+        # return UEFI variable at index into tree
+        if index < 0:
+            raise Exception("Bad index passed to get_var_by_index")
+
+        internal_idx = -1
+        # caller expects function to have variable set to None once list is exhausted
+        variable = None
+        name = None
+        exec = None
+
+        for item in self._cfg_list:
+            if item['type'] == 'Reserved':
+                continue
+
+            item = self.locate_cfg_item(item['path'])
+            if item is None:
+                raise Exception("Failed to locate item from path: %s" % item['path'])
 
-        # self.write_delta_file (delta_file, platform_id, lines)
-        return (execs, bytes_array)
+            exec = self.locate_exec_from_item(item)
+
+            if exec is None:
+                raise Exception("Failed to locate exec from item: %s" % item['path'])
+
+            # we have the exec, this is a real config item
+            internal_idx += 1
+            if internal_idx == index:
+                break
+
+        if internal_idx < index:
+            return None, None
+
+        bytes = self.get_field_value(exec)
+        offset = 0
+        offset += int(exec['CfgHeader']['length'], 0)
+        offset += int(exec['CondValue']['length'], 0)
+        tag_val = cfghdr_to_value(exec['CfgHeader']['value'])
+        name = "Device.ConfigData.TagID_%08X" % int(tag_val, 16)
+        variable = create_vlist_buffer(
+            UEFIVariable(name, uuid.UUID(SETUP_CONFIG_POLICY_VAR_GUID), bytes[offset:], attributes=3)
+        )
+
+        return variable, name
+
+    def generate_var_list(self):
+        varlist = b''
+        found_list = []
+        for item in self._cfg_list:
+            item = self.locate_cfg_item(item['path'])
+            if item is None:
+                raise Exception("Failed to locate item from path: %s" % item['path'])
+
+            exec = self.locate_exec_from_item(item)
+
+            # there are three cfg items associated with each exec
+            if exec is None or exec["CfgHeader"]["indx"] in found_list:
+                continue
+
+            found_list.append(exec["CfgHeader"]["indx"])
+
+            bytes = self.get_field_value(exec)
+            offset = 0
+            offset += int(exec['CfgHeader']['length'], 0)
```

Comment:
Does this already take into account the dword to byte change?

Reply:
This should be agnostic of the dword to byte change, since we are just reading the length out of the structure. When it was dwords, we'd have a different length, but from the outside, all we care about is passing over the header, which we trust the length to have been set correctly.